packages/relay/src/lib/services/metricService/metricService.ts

```diff
@@ -0,0 +1,345 @@
+          status,
+          requestId,
+        } as IExecuteQueryEventPayload);
+      }
+
+      if (txRecordChargeAmount !== 0) {
+        this.addExpenseAndCaptureMetrics({
+          executionType: ExecutionType.RECORD_QUERY_EXECUTION,
+          transactionId,
+          txConstructorName,
+          callerName,
+          cost: txRecordChargeAmount,
+          gasUsed: 0,
+          interactingEntity,
+          status,
+          requestId,
+        } as IExecuteQueryEventPayload);
+      }
+    }
+  }
+
+  /**
+   * Adds the expense to the HBAR rate limiter and captures the relevant metrics for the executed transaction.
+   *
+   * @param {IExecuteQueryEventPayload} payload - The payload object containing details about the transaction.
+   * @param {string} payload.executionType - The type of execution (e.g., `TransactionExecution`, `TransactionRecordQuery`).
+   * @param {string} payload.transactionId - The unique identifier for the transaction.
+   * @param {string} payload.txConstructorName - The name of the transaction constructor.
+   * @param {string} payload.callerName - The name of the entity calling the transaction.
+   * @param {number} payload.cost - The cost of the transaction in tinybars.
+   * @param {number} payload.gasUsed - The amount of gas used during the transaction.
+   * @param {string} payload.interactingEntity - The entity interacting with the transaction.
+   * @param {string} payload.status - The entity interacting with the transaction.
+   * @param {string} payload.requestId - The unique identifier for the request.
+   * @returns {void} - This method does not return a value.
+   */
+  public addExpenseAndCaptureMetrics = ({
+    executionType,
+    transactionId,
+    txConstructorName,
+    callerName,
+    cost,
+    gasUsed,
+    interactingEntity,
+    status,
+    requestId,
+  }: IExecuteQueryEventPayload): void => {
+    const formattedRequestId = formatRequestIdMessage(requestId);
+    this.logger.trace(
+      `${formattedRequestId} Capturing HBAR charged: executionType=${executionType} transactionId=${transactionId}, txConstructorName=${txConstructorName}, callerName=${callerName}, cost=${cost} tinybars`,
+    );
+
+    this.hbarLimiter.addExpense(cost, Date.now(), requestId);
+    this.captureMetrics(executionType, txConstructorName, status, cost, gasUsed, callerName, interactingEntity);
+  };
+
+  /**
+   * Retrieves the cost metric for consensus node client operations.
+   *
+   * @returns {Histogram} - The histogram metric tracking the cost of consensus node client operations.
+   */
+  public getCostMetric(): Histogram {
+    return this.consensusNodeClientHistogramCost;
+  }
+
+  /**
+   * Retrieves the gas fee metric for consensus node client operations.
+   *
+   * @returns {Histogram} - The histogram metric tracking the gas fees of consensus node client operations.
+   */
+  public getGasFeeMetric(): Histogram {
+    return this.consensusNodeClientHistogramGasFee;
+  }
```

Comment:
QQ: Are these getters used anywhere outside of the tests? We shouldn't expose things only because of tests. 

Let's keep those private and directly access them with `@ts-ignore` comments in the spec files.

Example usage inside `metricService.spec.ts`:
```
// @ts-ignore
const gasFeeHistogram =  metricService.consensusNodeClientHistogramGasFee;
```

Reply:
Fair. Updated.